walker.go

```diff
@@ -231,14 +231,23 @@ func (w *Walker) worker(ctx context.Context, chPaths <-chan string) error {
 		if err != nil {
 			return fmt.Errorf("unable to get file stat on base path %q: %v", path, err)
 		}
-		if err := filepath.Walk(path, func(p string, info os.FileInfo, err error) error {
-			p = NormalizePath(p, info.IsDir())
+		if err := filepath.WalkDir(path, func(p string, d os.DirEntry, err error) error {
+			// If the initial stat of the root dir fails we can get a nil value for d along with the
+			// PathError from os.Stat. See WalkDirFunc for details.
+			if d == nil && err != nil {
```

Comment:
My understanding is that d can only be nil if err != nil.

AFAIK `if err != nil {` should be enough.

I don't mind programming even more defensively, but in that case, we should use `||` and not `&&`. Thoughts?

Reply:
The [godoc for filepath.WalkDirFunc](https://pkg.go.dev/io/fs#WalkDirFunc) states that `d == nil && err != nil` is a different case from `d != nil && err != nil`. The first occurs when `os.Stat` on the _root_ dir of a walk fails, the second is when a `ReadDir` fails on any directory.

This is meant to catch the first case and let the second be handled by the existing error check. Given they fail on different operations I figured different errors was appropriate.